packages/flutter_tools/lib/src/fuchsia/application_package.dart

```diff
@@ -34,6 +34,7 @@ abstract class FuchsiaApp extends ApplicationPackage {
     }
     return PrebuiltFuchsiaApp(
       farArchive: applicationBinary.path,
+      applicationPackage: applicationBinary,
```

Comment:
Same here, we can just pass in `applicationBinary` and get the path off it later instead of storing both.
```suggestion
      applicationPackage: applicationBinary,
```